spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/embedded/EmbeddedLdapAutoConfiguration.java

```diff
@@ -89,6 +90,7 @@ public EmbeddedLdapAutoConfiguration(EmbeddedLdapProperties embeddedProperties)
 	@Bean
 	@DependsOn("directoryServer")
 	@ConditionalOnMissingBean
+	@ConditionalOnClass(ContextSource.class)
```

Comment:
Unfortunately, that shouldn't fix the underlying problem. When the JVM loads the class, it will parse method signature and will fail anyway, see the javadoc of `@ConditionalOnClass` for more details.

Reply:
I gotcha. Yes, that totally makes sense. Cause for continued fail and the inner class as a solution. Will make that change.
I'll also take a look at writing a test for this. Always wondered how this would be testable. Lots to learn here.